### ISSUE TITLE
does not start at all

### ISSUE THREAD

**Issue by mxdpeep** [ValveSoftware/steam-for-linux#8222]
Running Steam on ubuntu 18.04 64-bit
STEAM_RUNTIME is enabled automatically
Steam runtime environment up-to-date!
Steam client's requirements are satisfied
WARNING: Using default/fallback debugger launch
/home/mxdpeep/.local/share/Steam/ubuntu12_32/steam

that's all I can see, nothing more happens

**Comment by kisak-valve** [comment]
Hello @mxdpeep, can you check if any 32 bit OpenGL application like `glxinfo -B` from `mesa-utils:i386` works? Also, what video card / drivr version are you using?

**Comment by scanno** [comment]
Got the same problem right now.

output from glxinfo -B:
`name of display: :1
display: :1  screen: 0
direct rendering: Yes
Memory info (GL_NVX_gpu_memory_info):
    Dedicated video memory: 6144 MB
    Total available memory: 6144 MB
    Currently available dedicated video memory: 5245 MB
OpenGL vendor string: NVIDIA Corporation
OpenGL renderer string: NVIDIA GeForce RTX 3060 Laptop GPU/PCIe/SSE2
OpenGL core profile version string: 4.6.0 NVIDIA 495.44
OpenGL core profile shading language version string: 4.60 NVIDIA
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile

OpenGL version string: 4.6.0 NVIDIA 495.44
OpenGL shading language version string: 4.60 NVIDIA
OpenGL context flags: (none)
OpenGL profile mask: (none)

OpenGL ES profile version string: OpenGL ES 3.2 NVIDIA 495.44
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.20
`

**Comment by kisak-valve** [comment]
Hello @scanno, are you sure that you tested a 32 bit variant of glxinfo? Unfortunately, Ubuntu doesn't provide a 32 bit builds of this package on 20.10, so maybe try unpacking https://packages.ubuntu.com/bionic/i386/mesa-utils/download with an archive utility and using the utility without installing it to the system. Also, please check if you have nVidia's 32 bit userspace libraries installed with something like `apt policy libnvidia-gl-495 libnvidia-gl-495:i386`.

**Comment by scanno** [comment]
> Hello @scanno, are you sure that you tested a 32 bit variant of glxinfo? Unfortunately, Ubuntu doesn't provide a 32 bit builds of this package on 20.10, so maybe try unpacking https://packages.ubuntu.com/bionic/i386/mesa-utils/download with an archive utility and using the utility without installing it to the system. Also, please check if you have nVidia's 32 bit userspace libraries installed with something like `apt policy libnvidia-gl-495 libnvidia-gl-495:i386`.

Yeah, I already checked that:
```
libnvidia-gl-495/impish-updates,impish-security,now 495.44-0ubuntu0.21.10.1 i386 [installed,automatic]
```

**Comment by scanno** [comment]
Weird, when i remove the ~/.steam directory, steam re-installs and I can use steam again. But starting a second time after logging out, fails again.

**Comment by scanno** [comment]
Ok some more info:
Running steam from the install generated .desktop file does not work.
Running steam from a shell worked.
When steam is started using the .desktop entry, it fails to start and also from the shell fails.
Deleting ~/.steam/steam.pid and then starting steam from the shell, starts steam.

So I don't know what is going on here.

**Comment by kisak-valve** [comment]
@scanno, it might be worthwhile to give #8179 a read and see if that's the issue you're seeing.

**Comment by scanno** [comment]
> @scanno, it might be worthwhile to give #8179 a read and see if that's the issue you're seeing.

Thanks setting PrefersNonDefaultGPU to false works.

**Comment by TheShadowOfHassen** [comment]
I have the same problem. I  upgraded to Ubuntu 21.10 and the error happened after I reinstalled my graphic drivers. I originally thought it was a bad upgrade, but when I reinstalled Ubuntu and steam I still had the problem.
Steam ran through the terminal once I installed it, but then didn’t when I ran it through the icon afterwards. It worked again through the terminal when I ran ‘sudo apt-get install libnvidia-gl-390:i386’ as some forums said, but when I restarted the computer again and ran it with the icon, it didn’t.
So next I ran it in the terminal and gave the exact same message as mxdpeep’s except the first line states that I am running ubuntu 21.10. I tried reinstalling steam but it did not work.
So now the only way I can get it to run is to delete the entire .steam folder, in the home directory, then run it through termninal. It will print out this:
(username the stand in for my name which is the stand in of the home directory )

Running Steam on ubuntu 21.10 64-bit
STEAM_RUNTIME is enabled automatically
Steam runtime environment up-to-date!
Steam client's requirements are satisfied
WARNING: Using default/fallback debugger launch
/home/username/.local/share/Steam/ubuntu12_32/steam
[2021-11-13 14:42:06] Startup - updater built Oct 13 2021 19:47:06
Installing breakpad exception handler for appid(steam)/version(1634158817)
[2021-11-13 14:42:06] Loading cached metrics from disk (/home/username/.local/share/Steam/package/steam_client_metrics.bin)
[2021-11-13 14:42:06] Using the following download hosts for Public, Realm steamglobal
[2021-11-13 14:42:06] 1. https://cdn.cloudflare.steamstatic.com, /client/, Realm 'steamglobal', weight was 100, source = 'update_hosts_cached.vdf'
[2021-11-13 14:42:06] 2. https://cdn.akamai.steamstatic.com, /client/, Realm 'steamglobal', weight was 100, source = 'update_hosts_cached.vdf'
[2021-11-13 14:42:06] 3. http://media.steampowered.com, /client/, Realm 'steamglobal', weight was 1, source = 'baked in'
Installing breakpad exception handler for appid(steam)/version(1634158817)
[2021-11-13 14:42:06] Verifying installation...
[2021-11-13 14:42:06] Verification complete
Loaded SDL version 2.0.17-6744061
Gtk-Message: 14:42:07.023: Failed to load module "gail"
Gtk-Message: 14:42:07.024: Failed to load module "atk-bridge"

(steam:66375): Gtk-WARNING **: 14:42:07.026: Unable to locate theme engine in module_path: "adwaita",
/usr/share/themes/Yaru-dark/gtk-2.0/main.rc:775: error: unexpected identifier 'direction', expected character '}'

(steam:66375): Gtk-WARNING **: 14:42:07.028: Unable to locate theme engine in module_path: "adwaita",
/usr/share/themes/Yaru-dark/gtk-2.0/hacks.rc:28: error: invalid string constant "normal_entry", expected valid string constant
Installing breakpad exception handler for appid(steam)/version(1634158817)
Runtime for steamwebhelper: defaulting to /home/username/.local/share/Steam/ubuntu12_64/steam-runtime-heavy
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
CAppInfoCacheReadFromDiskThread took 25 milliseconds to initialize
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Installing breakpad exception handler for appid(steam)/version(1634158817)
Xlib:  extension "NV-GLX" missing on display ":1".
MESA-INTEL: warning: Haswell Vulkan support is incomplete
Xlib:  extension "NV-GLX" missing on display ":1".
MESA-INTEL: warning: Haswell Vulkan support is incomplete 

It then gives me the login screen. Once I log into steam loads with all my games.  If I quit steam it will run again if I use the terminal. If I click the steam icon nothing will happen. If I run it with the terminal, it will give the same output as before when it did not work.  To get it to work again, I have to delete .steam folder in my home directory again and run it with my terminal.

If I delete the .steam folder again and run the steam with the icon in the start menu, it creates another .steam folder but steam will not work.

I tried the fix for error #8179 but it did not work.

In summary, if I delete the .steam folder in home and then only run steam through the terminal it will work, but if the desktop icon is hit, it will not until the .steam folder and home is deleted.

Thanks for the help!

**Comment by kisak-valve** [comment]
Hello @TheShadowOfHassen, please open a separate issue report for the Steam devs to ponder your issue.

Without more information from @mxdpeep, this issue report is not actionable. In general, we've seen Steam stall like that when the user is using the nVidia proprietary driver, but part of the driver has mismatched versions. The nVidia proprietary driver requires that the kernel module loaded into memory, 32 bit userspace libraries, and 64 bit userspace libraries are installed and have exactly the same version. When one of these versions are mismatched, then any OpenGL application can stall on initialization while waiting on the video driver.
